### PR TITLE
NF: browser use two columns explicitly

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
I believe this version is more readable.

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
works for me, thanks

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -624,12 +620,14 @@ open class CardBrowser :
...
         val sflRelativeFontSize = preferences.getInt("relativeCardBrowserFontSize", DEFAULT_FONT_SIZE_RATIO)
         val sflCustomFont = preferences.getString("browserEditorFont", "")
-        val columnsContent = arrayOf(COLUMN1_KEYS[mColumn1Index], COLUMN2_KEYS[mColumn2Index])
         // make a new list adapter mapping the data in mCards to column1 and column2 of R.layout.card_item_browser
         mCardsAdapter = MultiColumnListAdapter(
             this,
             R.layout.card_item_browser,
-            columnsContent, intArrayOf(R.id.card_sfld, R.id.card_column2),
+            COLUMN1_KEYS[mColumn1Index],
+            COLUMN1_KEYS[mColumn2Index],
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:628]: Duplicated column here where the original code used  COLUMN1_KEYS and COLUMN2_KEYS.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 2]
```diff
@@ -2112,12 +2116,14 @@ open class CardBrowser :
...
             val card = mCards[position]
-            (v.tag as Array<*>)
-                .forEachIndexed { i, col ->
-                    setFont(col as TextView) // set font for column
-                    col.text = card.getColumnHeaderText(fromKeys[i]) // set text for column
-                }
+            setFont(columnOneView)
+            columnOneView.text = card.getColumnHeaderText(columnOne)
+            setFont(columnTwoView)
+            columnOneView.text = card.getColumnHeaderText(columnTwo)
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:2126]: Duplicated code from two lines above with column one, should have been `columnTwoView.text`.

**Review by lukstbit — CHANGES_REQUESTED** [review]
Changes break functionality in CardBrowser by not showing the answer(the second column), see images below(first images shows the current main branch, second image shows CardBrowser with the changes from this PR):



Most likely related to the two issues below.

**Review by david-allison — COMMENTED** [review]
At least personally, I'd like to move to a CardBrowser which has more than two columns in the future.

**Comment by mikehardy** [comment]
> I'd like to move to a CardBrowser which has more than two columns in the future. I'm not sure if this is a shared opinion

Maybe if it was configurable? Otherwise I am thinking you have a bigger phone and keener eyesight then I do, good sir

**Comment by david-allison** [comment]
> > I'd like to move to a CardBrowser which has more than two columns in the future. I'm not sure if this is a shared opinion
> 
> Maybe if it was configurable? Otherwise I am thinking you have a bigger phone and keener eyesight then I do, good sir

We also support tablets

**Comment by mikehardy** [comment]
Those are use cases I could see it making sense but 2 still seems like a nice starter default. Maybe this is one of the rare-ish places where having different layouts for width >= 600dbp really does make sense. I do like responsive layouts a lot

**Comment by david-allison** [comment]
I fear I misspoke. For phones, I want 2 columns as a default.

For tablets, we can calculate this and may want more than two.

I don't think that fixing the number of columns to 2 brings us closer to this goal, so I don't agree with the premise of this PR

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:2126]: Thanks a lot!

**Comment by Arthur-Milchior** [comment]
I can understand having many columns. I believe that this will need so much rewritting that this PR don't really oppose it, because anyway, even current code would be hard to directly use to add more columns. Because size two is already hard coded here and both columns are distinct

**Comment by mikehardy** [comment]
I agree with both - column flexibility in future would be good but if I remember correctly from my review here, this is just making the previous hard code of 2 really obvious. Not a bad thing

**Comment by david-allison** [comment]
I don't like this for reasons stated above, but won't block it going in

**Comment by mikehardy** [comment]
Okay - may be sideways motion from a flexible layout perspective, but I don't think it hurts and it did seem like at least a marginal improvement

**Pull request merged**